src/groupeddataframe/splitapplycombine.jl

```diff
@@ -950,12 +955,31 @@ function _combine_with_first!(first::Union{AbstractDataFrame,
     starts = gd.starts
     ends = gd.ends
     # Handle first group
-    j = append_rows!(first, outcols, colstart, colnames)
-    @assert j === nothing # eltype is guaranteed to match
-    append!(idx, Iterators.repeated(gdidx[starts[rowstart]], _nrow(first)))
+
+    @assert isempty(colnames) == (_ncol(first) == 0) == isempty(outcols)
```

Comment:
`colnames` and `outcols` are guaranteed to have the same length thanks to the signature, so you can omit the last check. Actually you could just check `_ncol(first) == N`?

Reply:
right - this check is left from vector version.